osu.Game/Screens/Edit/Setup/ResourcesSection.cs

```diff
@@ -40,93 +39,110 @@ private void load()
         {
             Children = new Drawable[]
             {
-                backgroundTextBox = new FileChooserLabelledTextBox(".jpg", ".jpeg", ".png")
+                backgroundChooser = new LabelledFileChooser(".jpg", ".jpeg", ".png")
                 {
                     Label = "Background",
                     FixedLabelWidth = LABEL_WIDTH,
-                    PlaceholderText = "Click to select a background image",
-                    Current = { Value = working.Value.Metadata.BackgroundFile },
                     TabbableContentContainer = this
                 },
-                audioTrackTextBox = new FileChooserLabelledTextBox(".mp3", ".ogg")
+                audioTrackChooser = new LabelledFileChooser(".mp3", ".ogg")
                 {
                     Label = "Audio Track",
                     FixedLabelWidth = LABEL_WIDTH,
-                    PlaceholderText = "Click to select a track",
-                    Current = { Value = working.Value.Metadata.AudioFile },
                     TabbableContentContainer = this
                 },
             };
 
-            backgroundTextBox.Current.BindValueChanged(backgroundChanged);
-            audioTrackTextBox.Current.BindValueChanged(audioTrackChanged);
+            if (!string.IsNullOrEmpty(working.Value.Metadata.BackgroundFile))
+                backgroundChooser.Current.Value = new FileInfo(working.Value.Metadata.BackgroundFile);
+
+            if (!string.IsNullOrEmpty(working.Value.Metadata.AudioFile))
+                audioTrackChooser.Current.Value = new FileInfo(working.Value.Metadata.AudioFile);
+
+            backgroundChooser.Current.BindValueChanged(backgroundChanged, true);
+            audioTrackChooser.Current.BindValueChanged(audioTrackChanged, true);
         }
 
-        public bool ChangeBackgroundImage(string path)
+        public bool ChangeBackgroundImage(FileInfo source)
         {
-            var info = new FileInfo(path);
-
-            if (!info.Exists)
+            if (!source.Exists)
                 return false;
 
             var set = working.Value.BeatmapSetInfo;
 
+            var destination = new FileInfo($@"bg{source.Extension}");
+
             // remove the previous background for now.
             // in the future we probably want to check if this is being used elsewhere (other difficulties?)
-            var oldFile = set.Files.FirstOrDefault(f => f.Filename == working.Value.Metadata.BackgroundFile);
+            var oldFile = set.Files.FirstOrDefault(f => Path.GetFileNameWithoutExtension(f.Filename) == destination.Name);
```

Comment:
Doesn't this assume that the old background can never be named anything but `bg.jpg` or similar? I don't think this is a safe assumption to make. It certainly doesn't hold for several existing maps I have imported in my database.

This was more correct before, I'd revert this particular change. Enforcing the new filename is OK but the old one shouldn't be assumed to be matching `bg.jpg` or whatever. Same thing goes for the mirror change in `ChangeAudioTrack()`.